lib/scope.js

```diff
@@ -1133,13 +1319,20 @@ module.exports = {
     let matches = await scope.getMatchingRow( scope, { page_data: page_data, story_table: var_data });
     if ( env_vars.DEBUG ) {
       console.log( `~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~` );
-      console.log( `matches: ${ JSON.stringify( matches )}` );
+      console.log( `page_data:\n${ JSON.stringify( page_data )}`);
+      console.log( `matches:\n${ JSON.stringify( matches )}` );
     }
 
     let fields_to_set = [];
     for ( let row of matches ) {
       let field = row.field;
 
+      // Only wait for `<canvas>`. It takes a while to load, but will never be
+      // hidden. Other elements could be hidden, though, and will never show up.
+      if ( field.tag === 'canvas' ) {
+        await scope.page.waitForSelector('#dasigcanvas');
+      }
+
```

Comment:
Signature stuff. What we're doing here is getting the handle to send to `.setVariable()`. To get the handle for the canvas, we have to wait. I'm debating putting this in `.setVariable()` instead. If we want to start removing puppeteer from this function, that might help. We haven't yet assessed whether that's possible or not, though.

Reply:
It just occurred to me this may affect other things that use `.setVariable()`. Shoot. Not enough bandwidth for this...

Reply:
I can't find anything else using `.setVariable()`, so we should be safe.